packages/babel-plugin-htm/index.mjs

```diff
@@ -85,48 +77,49 @@ export default function htmBabelPlugin({ types: t }, options = {}) {
 			child = child.trim();
 		}
 		if (typeof child==='string') {
-			const matches = child.match(/\$\$\$_h_\[(\d+)\]/);
-			if (matches) return currentExpressions[matches[1]];
 			return stringValue(child);
 		}
 		return child;
 	}
 
 	function h(tag, props) {
 		if (typeof tag==='string') {
-			const matches = tag.match(/\$\$\$_h_\[(\d+)\]/);
-			if (matches) tag = currentExpressions[matches[1]];
-			else tag = t.stringLiteral(tag);
+			tag = t.stringLiteral(tag);
 		}
 
-		//const propsNode = props==null || Object.keys(props).length===0 ? t.nullLiteral() : t.objectExpression(
-		const propsNode = t.objectExpression(
-			Object.keys(props).map(key => {
-				let value = props[key];
-				if (typeof value==='string') {
-					const tokenizer = /\$\$\$_h_\[(\d+)\]/g;
-					let token, lhs, root, index=0, lastIndex=0;
-					const append = expr => {
-						if (lhs) expr = t.binaryExpression('+', lhs, expr);
-						root = lhs = expr;
-					};
-					while ((token = tokenizer.exec(value))) {
-						append(t.stringLiteral(value.substring(index, token.index)));
-						append(currentExpressions[token[1]]);
-						index = token.index;
-						lastIndex = tokenizer.lastIndex;
-					}
-					if (lastIndex < value.length) {
-						append(t.stringLiteral(value.substring(lastIndex)));
+		let propsNode;
+
+		if (t.isObjectExpression(props)) {
```

Comment:
I suppose this block is here because the `Object.assign` spreads that `htm` produces smushes together both objects and Babel nodes? PR #44 is a suggestion (is a bit extensive, though) for avoiding this.